python-avd/pyavd/_eos_designs/structured_config/underlay/router_pim_sparse_mode.py

```diff
@@ -28,37 +28,37 @@ def router_pim_sparse_mode(self: AvdStructuredConfigUnderlay) -> dict | None:
 
         Used for to configure multicast RPs for the underlay
         """
-        if self.shared_utils.underlay_multicast_rps is None:
+        if not self.shared_utils.underlay_multicast or not self.inputs.underlay_multicast_rps:
             return None
 
         rp_addresses = []
         anycast_rps = []
-        for rp_entry in self.shared_utils.underlay_multicast_rps:
-            rp_address = {"address": rp_entry["rp"]}
-            if rp_entry.get("groups") is not None:
-                if (acl := rp_entry.get("access_list_name")) is not None:
-                    rp_address["access_lists"] = [acl]
+        for rp_entry in self.inputs.underlay_multicast_rps:
+            rp_address = {"address": rp_entry.rp}
+            if rp_entry.groups:
+                if rp_entry.access_list_name:
+                    rp_address["access_lists"] = [rp_entry.access_list_name]
                 else:
-                    rp_address["groups"] = rp_entry["groups"]
+                    rp_address["groups"] = rp_entry.groups
 
             rp_addresses.append(rp_address)
 
-            if (nodes := get(rp_entry, "nodes")) is None or len(nodes) < 2:
+            if len(rp_entry.nodes) < 2:
                 continue
 
-            if get_item(nodes, "name", self.shared_utils.hostname) is None:
+            if self.shared_utils.hostname not in rp_entry.nodes:
                 continue
```

Comment:
same could group the two checks together

Reply:
done (all three)